lib/ssh_scan/database.rb

```diff
@@ -0,0 +1,26 @@
+# API Databases
+require 'ssh_scan/database/mongo'
+require 'ssh_scan/database/sqlite'
+
+module SSHScan
+  class DatabaseConfig
+    def initialize(opts = {})
+      @@db_type = opts['database']['type'] || 'mongodb'
```

Comment:
It should definitely work with instance variables, I would try and again and if it doesn't work mention me and point me to the code and I can help you debug.